crates/services/upgradable-executor/wasm-executor/src/ext.rs

```diff
@@ -148,8 +151,13 @@ pub fn input(size: usize) -> anyhow::Result<InputDeserializationType> {
 }
 
 /// Gets the next transactions by using the host function.
-pub fn next_transactions(gas_limit: u64) -> anyhow::Result<Vec<MaybeCheckedTransaction>> {
-    let next_size = unsafe { host::peek_next_txs_size(gas_limit) };
+pub fn next_transactions(
+    gas_limit: u64,
+    block_transaction_size_limit: u64,
+) -> anyhow::Result<Vec<MaybeCheckedTransaction>> {
+    // TODO[RC]: Peek, check size. If fits, continue. If too large, peek next (?)
+    let next_size =
+        unsafe { host::peek_next_txs_size(gas_limit, block_transaction_size_limit) };
```

Comment:
Note to self:
We should most likely keep the `peek_next_txs_size` and introduce new function for the limits.

Reply:
This has been taken care of by merging changes introduced here: https://github.com/FuelLabs/fuel-core/pull/2199